assets/js/src/experimental_features/experimental_features.jsx

```diff
@@ -7,24 +7,28 @@ import Notices from 'notices/notices.jsx';
 const ExperimentalFeatures = () => {
   const [flags, setFlags] = useState(null);
   const contextValue = useGlobalContextValue(window);
+  const [mounted, setMounted] = useState(false);
 
   useEffect(() => {
-    MailPoet.Ajax.post({
-      api_version: window.mailpoet_api_version,
-      endpoint: 'featureFlags',
-      action: 'getAll',
-    }).done((response) => {
-      const flagsMap = response.data.reduce((obj, item) => ({ ...obj, [item.name]: item }), {});
-      setFlags(flagsMap);
-    }).fail((response) => {
-      if (response.errors.length > 0) {
-        MailPoet.Notice.error(
-          response.errors.map((error) => error.message),
-          { scroll: true }
-        );
-      }
-    });
-  }, []);
+    if (!mounted) {
+      MailPoet.Ajax.post({
+        api_version: window.mailpoet_api_version,
+        endpoint: 'featureFlags',
+        action: 'getAll',
+      }).done((response) => {
+        const flagsMap = response.data.reduce((obj, item) => ({ ...obj, [item.name]: item }), {});
+        setFlags(flagsMap);
+      }).fail((response) => {
+        if (response.errors.length > 0) {
+          contextValue.notices.error(
+            <>{response.errors.map((error) => <p>{error.message}</p>)}</>,
+            { scroll: true }
+          );
+        }
+      });
+      setMounted(true);
+    }
+  }, [contextValue.notices, mounted]);
```

Comment:
Adding `mounted` here is kind of a hack. I tested this and was trying to understand why you do it this way (and @costasovo helped me to understand the problem) but I don't think this is a good solution. Now, this "mounted" workaround is used here, in a single place. However, looking at `useGlobalContextValue(...)`, there's a potential risk of doing this hack in many other components in the future.

Looking at how `useGlobalContextValue(...)` is implemented, it seems it always creates a new object and returns it. That's why this ends up in a loop (and adding a "mounted" state is a workaround to prevent that loop). My question is - why do we always return a new object?

Going deeper, the `useGlobalContextValue(...)` creates four other subcontexts, and those too are always new objects. Then `useEffect` with not only `contextValue` will loop but also a `useEffect` with `contextValue.notices` only. Moreover, this could have some memory-usage consequences - can't we end up creating many of these context objects over and over again for every single usage?

One more question concerning the `useGlobalContextValue(...)` - why do we have to pass `window` from every single component where we'll use it? Do you think it's a good idea that many components along our codebase will touch `window` object? I personally don't like it (in fact, I think `window` should be prohibited by eslint, with maybe one or a few exceptions).

Could you maybe explain those concerns a bit (and the design of this global context hook) in regard to the points raised above? If there's more to discuss it may worth a Slack chat with others as well. I think we should handle this first so we don't end up polluting our code with more occurrences of thins such as the "mounted".

It may as well worth explaining why we have some of the other contexts (for instance, segments/users context). I don't think it was really discussed and I'm not sure their existence is justifiable, those are static values that won't change and may be passed by props in many cases. How do we decide which data should be passed by props and which should have a context wrapper? Context may be overkill for some use cases, not to mention it's not a very intuitive construct. We should use it carefully.